objects/f42a3e/contained/a15998/script.lua

```diff
@@ -0,0 +1,418 @@
+        end
+        table.insert(result, " }\n")
+        return table.concat(result)
+    end
+
+    for boardGUID,objsData in pairs(output) do
+	local board = getObjectFromGUID(boardGUID)
+    for _,objData in pairs(objsData) do
+        local name = objData.name
+        local pos = objData.position
+        local state = objData.state
+        if not setupCoordsUnnamed[state] then
+            setupCoordsUnnamed[state] = {}
+            setupPiecesUnnamed[state] = {}
+            setupCoordsNamed[state] = {}
+            setupPiecesNamed[state] = {}
+        end
+        if name == "Empty Space" then
+            table.insert(setupCoordsUnnamed[state], "{x="..pos.x..", y="..pos.y..", z="..pos.z.."}")
+        else
+            table.insert(setupCoordsNamed[state], "{x="..pos.x..", y="..pos.y..", z="..pos.z.."}")
+            table.insert(setupPiecesNamed[state], "\""..name.."\"")
+        end
+    end
+    table.insert(boardLines, concat2DTable(combine2DTables(setupCoordsNamed, setupCoordsUnnamed), "posMap"))
+    table.insert(boardLines, concat2DTable(combine2DTables(setupPiecesNamed, setupPiecesUnnamed), "pieceMap"))
+    local boardScript = table.concat(boardLines, "\n").."\n"
+    board.setLuaScript(boardScript)
+	board.setLuaScript(boardScript)
+	--HACK: I have no clue why but it doesn't work when the script is applied only once :shrug:
+	board.reload()
+    setupCoordsUnnamed, setupPiecesUnnamed, setupCoordsNamed, setupPiecesNamed = {}, {}, {}, {}
+    end
+end
+
+function GetSpawnPositions()
+    local strifeablePieces = {
+      ["Explorer"] = true,
+      ["Town"] = true,
+      ["City"] = true
+    }
+    local boards = getMapTiles()
+
+    while true do
+        local moving = false
+        for _, obj in pairs(boards) do
+            if obj.isSmoothMoving() then
+                moving = true
+                break
+            end
+        end
+        if not moving then break end
+        coroutine.yield()
+    end
+
+    local output = {}
+    for _,board in pairs(boards) do
+        local boardTable = {}
+        local hits = Physics.cast({
+            origin = board.getPosition() + Vector(0, 0.45, 0),
+            direction = Vector(0, 1, 0),
+            type = 3,
+            size = board.getBounds().size,
+        })
+        for _,hit in pairs(hits) do
+            if not isIslandBoard({obj=hit.hit_object}) then
+                local subHits = Physics.cast({
+                    origin = hit.hit_object.getPosition() + Vector(0, 0.1, 0),
+                    direction = Vector(0, -1, 0),
+                    max_distance = 0.6,
+                })
+                local onBoard = false
+                for _,subHit in pairs(subHits) do
+                    if subHit.hit_object == board then
+                        onBoard = true
+                        break
+                    end
+                end
+                if onBoard then
+		    local state = hit.hit_object.getStateId()
```

Comment:
what is the purpose of state here?

Reply:
State is used to track what land the piece is in

Reply:
can you add a comment covering this?